modules/apps/journal/journal-web/src/test/java/com/liferay/journal/web/internal/dao/search/JournalRecentArticlesResultRowSplitterTest.java

```diff
@@ -0,0 +1,217 @@
+			0, localDateTime);
+		JournalArticle oneDayAgoJournalArticle = _getJournalArticle(
+			1, localDateTime);
+		JournalArticle sevenDaysAgoJournalArticle = _getJournalArticle(
+			7, localDateTime);
+		JournalArticle eightDaysAgoJournalArticle = _getJournalArticle(
+			8, localDateTime);
+		JournalArticle thirtyDaysAgoJournalArticle = _getJournalArticle(
+			30, localDateTime);
+		JournalArticle thirtyOneDaysAgoJournalArticle = _getJournalArticle(
+			31, localDateTime);
+
+		List<ResultRow> resultRows = Arrays.asList(
+			_getResultRow(todayJournalArticle),
+			_getResultRow(oneDayAgoJournalArticle),
+			_getResultRow(sevenDaysAgoJournalArticle),
+			_getResultRow(eightDaysAgoJournalArticle),
+			_getResultRow(thirtyDaysAgoJournalArticle),
+			_getResultRow(thirtyOneDaysAgoJournalArticle));
+
+		JournalRecentArticlesResultRowSplitter
+			journalRecentArticlesResultRowSplitter =
+				new JournalRecentArticlesResultRowSplitter(themeDisplay);
+
+		List<ResultRowSplitterEntry> resultRowSplitterEntries =
+			journalRecentArticlesResultRowSplitter.split(resultRows);
+
+		Assert.assertEquals(
+			resultRowSplitterEntries.toString(), 4,
+			resultRowSplitterEntries.size());
+
+		_assertResultRowSplitterEntry(
+			"today", resultRowSplitterEntries.get(0), themeDisplay,
+			todayJournalArticle);
+		_assertResultRowSplitterEntry(
+			"last-7-days", resultRowSplitterEntries.get(1), themeDisplay,
+			oneDayAgoJournalArticle, sevenDaysAgoJournalArticle);
+		_assertResultRowSplitterEntry(
+			"last-30-days", resultRowSplitterEntries.get(2), themeDisplay,
+			eightDaysAgoJournalArticle, thirtyDaysAgoJournalArticle);
+		_assertResultRowSplitterEntry(
+			"older", resultRowSplitterEntries.get(3), themeDisplay,
+			thirtyOneDaysAgoJournalArticle);
+	}
+
+	private void _assertResultRowSplitterEntry(
+		String key, ResultRowSplitterEntry resultRowSplitterEntry,
+		ThemeDisplay themeDisplay, JournalArticle... expectedJournalArticles) {
+
+		Assert.assertEquals(
+			_language.get(themeDisplay.getLocale(), key),
+			resultRowSplitterEntry.getTitle());
+
+		List<ResultRow> resultRows = resultRowSplitterEntry.getResultRows();
+
+		Assert.assertEquals(
+			resultRows.toString(), expectedJournalArticles.length,
+			resultRows.size());
+
+		List<JournalArticle> journalArticles = TransformUtil.transform(
+			resultRows, resultRow -> (JournalArticle)resultRow.getObject());
+
+		for (int i = 0; i < resultRows.size(); i++) {
+			ResultRow resultRow = resultRows.get(i);
+
+			Assert.assertEquals(journalArticles.get(i), resultRow.getObject());
+		}
+	}
+
+	private JournalArticle _getJournalArticle(
+		int days, LocalDateTime localDateTime) {
+
+		JournalArticle journalArticle = Mockito.mock(JournalArticle.class);
+
+		LocalDateTime modifiedLocalDateTime = localDateTime.minusDays(days);
+
+		Instant instant = modifiedLocalDateTime.toInstant(ZoneOffset.UTC);
+
+		Mockito.when(
+			journalArticle.getModifiedDate()
+		).thenReturn(
+			Date.from(instant)
```

Comment:
Maybe we could return here the  something like : Date.from(modifiedLocalDateTime.toInstant(ZoneOffset.UTC))